### ISSUE TITLE
white temperature slider not work with H6159

### ISSUE THREAD

**Issue by sagitt** [LaggAt/hacs-govee#2]
if i use slider on this model, i can see changes on the led strip but not for the temperature color, i see "less btightness intensity". this is the log:

[home-assistant.log](https://github.com/LaggAt/hacs-govee/files/5541069/home-assistant.log)

**Comment by LaggAt** [comment]
I just compared your commands and state results with mine, and they are just similar.

I see the line where we get the command in the integration, and the second line the library sends the command to the API.
```
[custom_components.govee.light] async_turn_on for Govee light XX:XX:XX:XX:XX:XX:XX:XX, kwargs: {'color_temp': 5362}
[govee_api_laggat.govee_api_laggat] control XX:XX:XX:XX:XX:XX:XX:XX: {'name': 'colorTem', 'value': 5362}
```
I also see the cache state containing the sent values, and - on yours and on mine - I see no color temperature in the results.

Difference seems - as we send and receive the same - you couldn't see a difference on your H6159.

Seems we cannot change color temperature using API on your device.
To test this assumption: could you turn off bluetooth (and when possible wifi and use just a mobile connection) on your phone and see if you can switch color temperature? The App *may* be albe, also we cannot, if it is using a different api or unknown commands. But if that doesn't work we know we are stuck when using API.

**Comment by sagitt** [comment]
> I just compared your commands and state results with mine, and they are just similar.
> 
> I see the line where we get the command in the integration, and the second line the library sends the command to the API.
> 
> ```
> [custom_components.govee.light] async_turn_on for Govee light XX:XX:XX:XX:XX:XX:XX:XX, kwargs: {'color_temp': 5362}
> [govee_api_laggat.govee_api_laggat] control XX:XX:XX:XX:XX:XX:XX:XX: {'name': 'colorTem', 'value': 5362}
> ```
> 
> I also see the cache state containing the sent values, and - on yours and on mine - I see no color temperature in the results.
> 
> Difference seems - as we send and receive the same - you couldn't see a difference on your H6159.
> 
> Seems we cannot change color temperature using API on your device.
> To test this assumption: could you turn off bluetooth (and when possible wifi and use just a mobile connection) on your phone and see if you can switch color temperature? The App _may_ be albe, also we cannot, if it is using a different api or unknown commands. But if that doesn't work we know we are stuck when using API.

disabling BT (see the text in the goove app confirming this) and using mobile connection (no wifi) the temperature works fine!

**Comment by sagitt** [comment]
i received this with my api key 
[Govee API Reference.pdf](https://github.com/LaggAt/hacs-govee/files/5541355/Govee.API.Reference.pdf)

**Comment by LaggAt** [comment]
The API reference matches mine, so I assume the app is doing something different.
Again to proove this theory let's put this to a test:

Could you try calling the API manually?
Here is a command you can try using a linux/mac box, if you are on windows i would recommend Postman.

```
curl --location --request PUT 'https://developer-api.govee.com/v1/devices/control' \
--header 'Govee-API-Key: INSERT_YOUR_API_KEY_HERE' \
--header 'Content-Type: application/json' \
--data-raw '{
    "device": "40:83:FF:FF:FF:FF:FF:FF",
    "model": "H6159",
    "cmd": {
        "name": "colorTem",
        "value": 2000
    }
}'
```

If the API works on this, you should be able to set the color temp to 2000 (see 'value' above) with this command.
This is basically what the client does (and the log shows it) - so if you get a command working we need to compare this command to the one the client library is creating.

**Comment by sagitt** [comment]
> The API reference matches mine, so I assume the app is doing something different.
> Again to proove this theory let's put this to a test:
> 
> Could you try calling the API manually?
> Here is a command you can try using a linux/mac box, if you are on windows i would recommend Postman.
> 
> ```
> curl --location --request PUT 'https://developer-api.govee.com/v1/devices/control' \
> --header 'Govee-API-Key: INSERT_YOUR_API_KEY_HERE' \
> --header 'Content-Type: application/json' \
> --data-raw '{
>     "device": "40:83:FF:FF:FF:FF:FF:FF",
>     "model": "H6159",
>     "cmd": {
>         "name": "colorTem",
>         "value": 2000
>     }
> }'
> ```
> 
> If the API works on this, you should be able to set the color temp to 2000 (see 'value' above) with this command.
> This is basically what the client does (and the log shows it) - so if you get a command working we need to compare this command to the one the client library is creating.

if i set manually from app "yellow temperature" using this command it set the strip to white color, but i tried to change the temperature value from 1000 to 8000 and the color is ever the same.

{"code":200,"message":"Success","data":{}}%     

this is the result

**Comment by LaggAt** [comment]
So the library does anything correct. 
I wonder a bit why you could send 1000 as this value is defined out-of-range in api description, I also do not get a success message when sending something outside of 2000-9000.

I discovered something unusual while trying here ... I could set color temperature once, where I land on the center of the color wheel selecting white. From that on nothing changes anymore.

I investigated a bit more, and the light does always select a white color with some low brightness, but the color temp doesn't change. So it seems you found a bug in the API noone have seen before.

I'll mail the API support for help. In the meanwhile I close here, as we cannot fix this here.
Feel free to also contact the API support on your own.

**Comment by LaggAt** [comment]
Mail to API support is out.

**Comment by sagitt** [comment]
> Mail to API support is out.

please give what i can write to the API support for have a similar technical request. thanks

**Comment by LaggAt** [comment]
Answer from Support:
We had fixed the issue you had, please try it again to check if you still have the issue.

I havn’t checked this as I am at work.

**Comment by sagitt** [comment]
> Answer from Support:
> We had fixed the issue you had, please try it again to check if you still have the issue.
> 
> I havn’t checked this as I am at work.

after mail color temperature works
but cold is hod and hot is cold :D

i think integration bug... with manual API call, 2000 is hot and 9000 is cold

**Comment by LaggAt** [comment]
yea, you 're right. Home Assistant expects mired, govee wants kelvin, and I did anything in kelvin. I'll fix this.

**Comment by sagitt** [comment]
> yea, you 're right. Home Assistant expects mired, govee wants kelvin, and I did anything in kelvin. I'll fix this.

well!

**Comment by clod986** [comment]
probably I have the same problem with H6003 but I'm a Home Assistant noob willing to help

@LaggAt how can I debug this to check if it can be solved with #33 and is not something else?

**Comment by LaggAt** [comment]
#33 isn't developed nor deployed yet, so you cant. Once it is developed, you could test it.
I mark this one here as duplicate.
